argoverse/data_loading/argoverse_stereo_loader.py

```diff
@@ -0,0 +1,433 @@
+                        log,
+                        "vehicle_calibration_stereo_info.json",
+                    )
+                )
+
+            self._log_list = [
+                x
+                for x in os.listdir(os.path.join(self.root_dir, "rectified_stereo_images_v1.1", self.split_name))
+                if valid_log(x)
+            ]
+
+        return self._log_list
+
+    @property
+    def image_list(self) -> Dict[str, List[str]]:
+        """return list of all image path (str) for the current log
+
+        Returns:
+            image_list: dictionary of list of image, with camera name as key
+        """
+        if self._image_list is None:
+            self._image_list = {}
+            for log in self.log_list:
+                self._image_list[log] = {}
+                for camera in RECTIFIED_STEREO_CAMERA_LIST:
+                    self._image_list[log][camera] = sorted(
+                        glob.glob(
+                            (
+                                os.path.join(
+                                    self.root_dir, "rectified_stereo_images_v1.1", self.split_name, log, camera, "*.jpg"
+                                )
+                            )
+                        )
+                    )
+                    self.image_count += len(self._image_list[log][camera])
+        return self._image_list[self.current_log]
+
+    @property
+    def disparity_list(self) -> Dict[str, List[str]]:
+        """return list of all image paths (str) for all cameras for the current log
+
+        Returns:
+            image_list: dictionary of list of image, with camera name as key
+        """
+        if self._disparity_list is None:
+            self._disparity_list = {}
+            for log in self.log_list:
+                self._disparity_list[log] = {}
+                STEREO_DISPARITY_LIST = ["stereo_front_left_rect_disparity", "stereo_front_left_rect_objects_disparity"]
+                for disparity in STEREO_DISPARITY_LIST:
+                    self._disparity_list[log][disparity] = sorted(
+                        glob.glob(
+                            (
+                                os.path.join(
+                                    self.root_dir, "disparity_maps_v1.1", self.split_name, log, disparity, "*.png"
+                                )
+                            )
+                        )
+                    )
+                    self.image_count += len(self._disparity_list[log][disparity])
+        return self._disparity_list[self.current_log]
+
+    @property
+    def image_timestamp_list(self) -> Dict[str, List[int]]:
+        """return dict of list of image timestamp (str) for the current log.
+
+        Returns:
+            image_timestamp_list: dictionary of list of image timestamp for all cameras
+        """
+        assert self.image_list is not None
+        assert self._image_list is not None
+
+        if self._image_timestamp_list is None:
```

Comment:
prefer to return early, rather than nest

Reply:
Sure, next commit will return early.